### PR TITLE
M4 Grouping and Aggregating lesson

### PR DESCRIPTION
**Pull request by juliet-e**
# Curriculum PR Template

### Does this PR close any issues?
#313 

### Description
Adds this mod 2 lesson adapted for Mod 4. Updates set list API to include a m2m complete branch that this class can start off of.

### What questions do you have/what do you want feedback on? (optional)
- quick read through to make sure it reads well and formatting looks ok
- sanity check on the answers to the practice problems below

### CODE REVIEW

**Comment by epintozzi** [comment]
Everything looks good to me. I pulled the branch to read through and walk through the exercises and all appears to be correct. 

Are you okay with us linking the Set List repo? I'm thinking it just makes it easier for us/students and not have to keep it updated separately in the calendar event.

File: module4/lessons/grouping_aggregating.md [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+
+PostgresSQL comes with some handy built-in aggregate functions. From the [Postgresql Docs](https://www.postgresql.org/docs/14/functions-aggregate.html): `Aggregate functions compute a single result from a set of input values`. Basically, aggregate functions will do some math for us.
+
+Take a minute to look through the [docs](https://www.postgresql.org/docs/14/functions-aggregate.html) and see what the DB can do for you.
+
+While we are talking about functions built in to Postgresql, it is important to note that nearly all relational database systems include some form of aggregating, for example [MySQL](https://dev.mysql.com/doc/refman/8.0/en/group-by-modifiers.html).
+
+### SQL Aggregates
+
+We'll be working with Set List on the `many-to-many-complete` branch.  Let's run `rails dbconsole` in SetList to open up a connection to our Postgres Database. Let's find the average length of all songs:
```
Comment (epintozzi) [inline module4/lessons/grouping_aggregating.md:36]: Perhaps here would be a good spot to link the repo if we're open to that idea?

Comment (memcmahon) [reply module4/lessons/grouping_aggregating.md]: Link has been added!

**Comment by memcmahon** [comment]
Agree with Erin - this looks good.  I just added the repo link and a terminal command for opening rails console.

**Pull request merged**